### PR TITLE
fix: cookies filter secure invalid

### PR DESCRIPTION
**Pull request by BlackHole1**
#### Description of Change

image

gist: https://gist.github.com/6d2ba80a835a75f90f385a2f4664a6a0

This is a simple judgment condition error that just needs to be corrected to the correct judgment condition.

#### Checklist

- [ ] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [ ] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Fixed an incorrect result returned when using secure as a cookies filter.

### CODE REVIEW

**Review by codebytere — CHANGES_REQUESTED** [review]
Can you please describe the issue in more detail as well as properly fill out the PR template? Additionally, we should have some tests for this.

**Comment by BlackHole1** [comment]
> Can you please describe the issue in more detail as well as properly fill out the PR template? Additionally, we should have some tests for this.

@codebytere Updated. The whole file does not have any unit tests at the moment. And the change in this PR is very simple, it is just if logic is written wrong.

**Comment by BlackHole1** [comment]
@codebytere Description updated and already add tests.

File: shell/browser/api/electron_api_cookies.cc [hunk 1]
```diff
@@ -128,7 +128,7 @@ bool MatchesCookie(const base::Value::Dict& filter,
       !MatchesDomain(*str, cookie.Domain()))
     return false;
   absl::optional<bool> secure_filter = filter.FindBool("secure");
-  if (secure_filter && *secure_filter == cookie.IsSecure())
+  if (secure_filter && *secure_filter != cookie.IsSecure())
     return false;
   absl::optional<bool> session_filter = filter.FindBool("session");
   if (session_filter && *session_filter != !cookie.IsPersistent())
```
Comment (deepak1556) [inline shell/browser/api/electron_api_cookies.cc:134]: Maybe also switch this for simplicity

```suggestion
  if (session_filter && *session_filter == cookie.IsPersistent())
```

**Review by deepak1556 — APPROVED** [review]

**Comment by deepak1556** [comment]
@BlackHole1 can you update the `Notes:` section with user facing release note since this is a bug fix.

**Comment by BlackHole1** [comment]
@deepak1556 Done. I updated the `notes` description.

**Review by zcbenz — APPROVED** [review]

**Review by codebytere — APPROVED** [review]

**Pull request merged**